### PR TITLE
feat(node-api): wire to backend

### PR DESCRIPTION
**Pull request by archbear**
## Summary by CodeRabbit

- **New Features**
  - Introduced a modular backend structure for improved state management and data handling.
  - Added a structured framework for managing various backend operations in the blockchain node.

- **Updates**
  - Updated dependencies to enhance compatibility and performance, including new libraries for web handling and data validation.
  - Revised Docker configuration for better module management and integration in containerized environments.

- **Improvements**
  - Enhanced configurability and modularity within the Node API framework, streamlining backend operations.

### CODE REVIEW

**Review by itsdevbear — COMMENTED** [review]
utack, good starting cleannup work

do we think there is a way to decouple handlers from echo?

File: mod/node-api/backend/storage/helpers.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package storage
+
+import (
+	"github.com/berachain/beacon-kit/mod/primitives/pkg/encoding/hex"
+)
+
+const (
```
Comment (itsdevbear) [inline mod/node-api/backend/storage/helpers.go:27]: we can probably make an api constants folder or something

File: mod/node-api/backend/beacon.go [hunk 2]
```diff
@@ -24,158 +24,190 @@ import (
...
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/crypto"
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/math"
 )
 
-func (h Backend) GetGenesis(ctx context.Context) (common.Root, error) {
+// GetGenesis returns the genesis state of the beacon chain.
+func (b Backend[
+	_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _,
+]) GetGenesis(ctx context.Context) (common.Root, error) {
 	// needs genesis_time and gensis_fork_version
```
Comment (itsdevbear) [inline mod/node-api/backend/beacon.go:38]: do we want to make a StateID -> BeaconState Middleware thing? cause this code is gonna be repeated over and over

Comment (archbear) [reply mod/node-api/backend/beacon.go:38]: thats kinda my idea with storage/backend but i didnt really know where to put that

File: mod/config/go.mod [hunk 3]
```diff
@@ -72,7 +79,12 @@ require (
 	github.com/klauspost/cpuid/v2 v2.2.8 // indirect
 	github.com/kr/pretty v0.3.1 // indirect
 	github.com/kr/text v0.2.0 // indirect
+	github.com/labstack/echo/v4 v4.12.0 // indirect
```
Comment (itsdevbear) [inline mod/config/go.mod:82]: config coupled to web framework

File: mod/node-api/go.mod [hunk 4]
```diff
@@ -2,23 +2,51 @@ module github.com/berachain/beacon-kit/mod/node-api
...
 	github.com/labstack/echo/v4 v4.12.0
 	github.com/stretchr/testify v1.9.0
 )
 
 require (
+	buf.build/gen/go/cometbft/cometbft/protocolbuffers/go v1.34.2-20240312114316-c0d3497e35d6.2 // indirect
+	buf.build/gen/go/cosmos/gogo-proto/protocolbuffers/go v1.34.2-20240130113600-88ef6483f90f.2 // indirect
+	cosmossdk.io/api v0.7.5 // indirect
+	cosmossdk.io/collections v0.4.0 // indirect
+	cosmossdk.io/core v0.12.1-0.20240623110059-dec2d5583e39 // indirect
```
Comment (itsdevbear) [inline mod/node-api/go.mod:29]: why does node-api need all the cosmos deps?

**Comment by calbera** [comment]
TODO: Investigate why first 7 (0x1, ..., 0x7) slots queried for return an empty slot root

**Pull request merged**